### PR TITLE
Initial steps to disentangle tensor tests

### PR DESCRIPTION
**Pull request by kshyatt**
The idea here is to use something like what we have in MAK to reduce the duplication across test files... if people like this approach I will continue on to `indexmanipulation` tests and others

### CODE REVIEW

**Comment by kshyatt** [comment]
This should also hopefully help with https://github.com/QuantumKitHub/TensorKit.jl/issues/428

**Review by lkdvos — COMMENTED** [review]
Definitely in favor of having a more modular and structured test set, but I was wondering if for the GPU tests we really want to repeat the tests in the same way we do them on CPU, and not just check that they give the same results as on CPU.

For example, a pattern I quite like is something like this:
https://github.com/QuantumKitHub/Strided.jl/blob/2a2b380e3718b2875c5125672fc9e58140c68e3b/test/gpu.jl#L1-L12

To make this more concrete, while I know that for CPU we test that `norm` is well-defined, i.e. `norm(2t) = 2norm(t)`, ... I would probably just check in the GPU case that `norm(t) = norm(adapt(CuVector, t))` and be done with it?

**Comment by kshyatt** [comment]
>  I was wondering if for the GPU tests we really want to repeat the tests in the same way we do them on CPU, and not just check that they give the same results as on CPU.

It's a fair question. For now, I'm keeping them mirrored because we ran into weirdness where `norm` specifically of an `AdjointTensorMap{<:CuTensorMap}` was doing something dumb... there were enough edge cases that I was biased towards keeping the bigger suite of tests. But maybe for CI it would make sense to have a slimmer version which just directly compares with the CPU in a more restricted set of cases?

File: test/testsuite/tensors/construction.jl [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        @test scalartype(t) == T
+        @test norm(t) == 0
+        @test codomain(t) == W
+        @test space(t) == (W ← one(W))
+        @test domain(t) == one(W)
+        @test typeof(t) == TensorMap{T, spacetype(t), 5, 0, AT}
+    end
+end
+
+function basic_blocks_properties(t, W)
```
Comment (Jutho) [inline test/testsuite/tensors/construction.jl:13]: Is it intentional that this one is not following the format

```julia
return @testset "Basic `blocks` properties" begin

end

Comment (kshyatt) [reply test/testsuite/tensors/construction.jl:13]: Ah, no 😓

**Comment by lkdvos** [comment]
> where `norm` specifically of an `AdjointTensorMap{<:CuTensorMap}` was doing something dumb...

I don't think this is really an argument in favor or against the different setups, you could also just do `compare(t -> norm(t'), ...)`.
I'm not saying to cut down on the things we test, I'm mostly advocating for not testing them by comparison to analytical results and just comparing with CPU results where possible.
